### ISSUE TITLE
(Update: 5 May 2023) BIOS setting PM L1 SS (Response from ASRock TSD)

### ISSUE THREAD

**Issue by LancerDawg75** [IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT#303]
### Checklist [README]

- [X] Device is not undervolted nor overclocked
- [X] Device is using the [latest drivers](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/wiki/Intel-driver-updates-and-changelogs)
- [X] Application is not cracked, modded and use the latest patch

### Application [Required]

Intel Graphics Command Centre and Arc Control

### Processor / Processor Number [Required]

AMD Ryzen 5800X3D

### Graphic Card [Required]

Intel Arc A770 Limited Edition

### GPU Driver Version [Required]

- [X] 31.0.101.4335
- [ ] 31.0.101.4314
- [ ] 31.0.101.4311
- [ ] 31.0.101.4257
- [ ] 31.0.101.4255
- [ ] 31.0.101.4148
- [ ] 31.0.101.4146
- [ ] 31.0.101.4125
- [ ] 31.0.101.4123
- [ ] 31.0.101.4091
- [ ] 31.0.101.4090
- [ ] 31.0.101.4034
- [ ] 31.0.101.4032
- [ ] 31.0.101.3975

### Rendering API [Required]

- [ ] Vulkan
- [ ] OpenGL
- [ ] DirectX12
- [ ] DirectX11
- [ ] DirectX10
- [ ] DirectX9
- [X] Not applicable

### Windows Build Number [Required]

- [X] Windows 11 22H2
- [ ] Windows 11 21H2
- [ ] Windows 10 22H2
- [ ] Windows 10 21H2
- [ ] Other (Please specify)

### Intel System Support Utility report

[SSU 1 May 2023.txt](https://github.com/IGCIT/Intel-GPU-Community-Issue-Tracker-IGCIT/files/11361932/SSU.1.May.2023.txt)

### Description and steps to reproduce [Required]

EDIT: To separate issues, and work on this a little more

Update: 1 May 2023

I decided to try to DDU the driver in safe mode, enable only IOMMU + DMA Kernel Protection, and reinstall the driver, and it posts normally. I really don't want to try the DMAr POST Protection because I think there might be a driver conflict in the Pre-Boot to Boot to Windows timeframe, but if you need me to recreate it, I will. 

There is part of me that thinks that maybe a setting that I had enabled in BIOS leftover from my previous GPU called PM L1 SS was enabled that might have caused an issue, but not totally sure as it seems to be working now with what I have kept enabled. So, if the moderators/owner of the site just wants to mark it as closed, then that would be okay. 

Update addendum - 3 May 2023

Trying to follow the instructions on setting a lower idle power consumption by [this Intel article
](https://www.intel.com/content/www/us/en/support/articles/000092564/graphics.html), it is the BIOS setting for PM L1 SS that is causing the card not to display anything. 

If any of the settings are enabled in BIOS for PM L1 SS (For my ASRock B550 PG Velocita it is found in Advanced > AMD PBS - along with the adjustments to PCIE lane multiplier and PCIE Gen adjustments)

- L 1.1
- L 1.2
- L 1.1 + 1.2 

The system will not show a BIOS POST screen, but will boot to Windows successfully as the Dr. Debug LEDs will shut off once a successful boot has been completed. Pressing the power button on the motherboard (or you could probably do it on the case too) will shut down Windows properly. For me, a CMOS Reset had to be performed each time I tried any of the PM L1 SS options and in this setting, it was the only setting I changed after the CMOS reset. 

After trying all the settings with an unsuccessful result, I restored all my previous BIOS settings with PM L1 SS disabled. The system successfully had an ASRock POST screen, would successfully boot to Windows, but with no display. 

I attempted to press the power button on the motherboard again to perform a proper shutdown of Windows. Restarted the system again. Again, the ASRock POST screen displayed, but this time, the system booted normally with Windows being displayed. I am not sure what the second power off did, but that helped it return to normal. 

I will be also sending a technical support ticket to ASRock concerning this issue as it might be motherboard related, but from my semi-educated guess, the BIOS and the Arc GOP do not like each other or whatever AMD and/or ASRock has implemented as I believe this is the first implementation of PM L1 SS that I have seen on the BIOS options screen (I am on the latest version for the B550 PG Velocita - version L 2.62, which brings us AGESA 1.2.0.8)

For now, applying the BIOS setting is not possible, but going into the Windows power plan and adjusting the PCI Express Link State Power Management to "Maximum Power Savings" has helped it so far in reducing the power draw (via GPU-Z) for me from 36-43W when doing nothing and not touching anything, down to 4-17W of draw. 

So, at the end of all this, I am concluding that it is not an IOMMU issue as I originally had thought, but the motherboard's ASPM settings in BIOS and possible conflict with the Arc's VBIOS....or it's just isolated to my system.

(Update 5 May 2023) Response from ASRock 

tl;dr - You don't need it as Windows handles it all. The setting ".....is a feature request from those who had the initial BIOS and want to try out the original BIOS options"

![ASRock TS1](https://user-images.githubusercontent.com/55825789/236277787-af6d121f-6e58-4253-a7a7-53d0c6a8dd74.jpg)

![ASRock TS2](https://user-images.githubusercontent.com/55825789/236277847-ea80ae16-c0d2-45ee-9877-22bc5c387b4a.jpg)

![ASRock TS3](https://user-images.githubusercontent.com/55825789/236278108-14d9d2d5-2e6a-44cc-b629-36b9546d62e0.jpg)

![ASRock TS4](https://user-images.githubusercontent.com/55825789/236279155-1b293d53-7867-4fed-9ab4-dd3ce1f7ece1.jpg)

I really didn't like the answer, but the tech support operative is technically correct. The power draw does go down from what used to be a range of around 37-44W power draw. So, I guess we call the issue closed? It -is- the setting in BIOS that Intel specifies, but by ASRock, it isn't or superfluous?

 
### Device / Platform

Not applicable

### Crash dumps [Required, if applicable]

Not applicable (I think) as the system does not crash and I am not seeing anything relevant (no Error or Warning) towards Intel Arc or graphics related issues during this testing

**Comment by IGCIT** [comment]
Hi,

yes, we prefer to have a github issue for each issue instead of multiple into one so that it is easier to keep track of them

about the rebar issue, this is known and you can join discussion here: #209

**Comment by LancerDawg75** [comment]
Thank you for your reply. Given this reply, if it is alright to edit this particular issue, I will:

- Remove the issue regarding Arc Control + ResizeBAR 
- Split the issues between two tickets:
 - Current Ticket: BIOS Settings and Windows boot issue
  - Next Ticket: Intel Graphics Command Center HDMI options issue

**Comment by IGCIT** [comment]
> Thank you for your reply. Given this reply, if it is alright to edit this particular issue, I will:
> 
>     * Remove the issue regarding Arc Control + ResizeBAR
> 
>     * Split the issues between two tickets:
> 
>     * Current Ticket: BIOS Settings and Windows boot issue
> 
>     * Next Ticket: Intel Graphics Command Center HDMI options issue

ok to all, thanks :)

**Comment by Arturo-Intel** [comment]
Hello @LancerDawg75,

Thank you for the details on this behavior, we are going to replicate this issue in our lab, i think we have an ASRock mobo somewhere here.

Either way, can you help me sharing your VBios version? 
you can get it from ARC Control > Settings gear > System info > IFWI (vBios)

--r2

**Comment by LancerDawg75** [comment]
IFWI (VBIOS) version 20.1053.0

**Comment by Arturo-Intel** [comment]
So if I understand correctly, Intel ARC was working (saving energy) as expected using the values provide by default
But the MoBo provides a way to change that value with AMD defaults that was affecting the GPU to wake up from Idle state, right?

Hmm curious case,
well if it is ok with you @LancerDawg75 we can close this case now
@IGCIT 

-- r2

**Comment by LancerDawg75** [comment]
I think maybe AMD’s substates were either too aggressive or just didn’t play well with the GOP because when enabled, the GPU was active as my system booted -normally-

(“Normally” in the realm that the motherboard BIOS or Arc’s VBIOS didn’t see a problem and so, let it boot to Windows. It just didn’t allow the GPU to display. I guess you could see that as leaving it in a perpetual L1 state of full sleep.

If you don’t see any issue with it, sir. 
I got my answer, more or less, and I guess - with testing on your end to see if it replicates on an AMD platform, that the only setting that should be applied is the PCI Express Link State Power Management to max power savings in the Windows power plans.
